### PR TITLE
spml_ucx: allow registering the heap in non-blocking mode.

### PR DESCRIPTION
**Pull request by yosefe**

### CODE REVIEW

**Comment by yosefe** [comment]
@alex-mikheev pls review

File: oshmem/mca/spml/ucx/spml_ucx_component.c [hunk 1]
```diff
@@ -101,6 +101,10 @@ static int mca_spml_ucx_component_register(void)
                                     "How may disconnects go in parallel",
                                     &mca_spml_ucx.num_disconnect);
 
+    mca_spml_ucx_param_register_int("heap_reg_nb", 0,
```
Comment (miked-mellanox) [inline oshmem/mca/spml/ucx/spml_ucx_component.c]: maybe by default ON?

Comment (alex-mikheev) [inline oshmem/mca/spml/ucx/spml_ucx_component.c]: What happens if the flag is not supported for the device? Will ucp_mem_map() fail ?

**Comment by miked-mellanox** [comment]
@yosefe - please port to v2.x

**Comment by yosefe** [comment]
@alex-mikheev if the flag is not supported t will have no effect.

**Pull request merged**